code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -775,15 +775,15 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		pai = C
 		user << "<span class='notice'>You slot \the [C] into [src].</span>"
 		updateUsrDialog()
-	else if(istype(C, /obj/item/weapon/pen))
-		var/obj/item/weapon/pen/O = locate() in src
-		if(O)
-			user << "<span class='warning'>There is already a pen in \the [src]!</span>"
+	else if(is_type_in_list(C, contained_item)) //Checks if there is a pen
+		if(inserted_item)
+			user << "<span class='warning'>There is already an item in \the [src]!</span>"

```

Comment:
There's an issue with putting [inserted_item] there. 
Instead of outputting: "There is already a pen in the PDA!"
It will instead output: "There is already the pen in the PDA!"